### DISCUSSION TITLE
Need help understanding Starship

### DISCUSSION THREAD

**Discussion by purezen** [starship/starship#4919] [Q&A]
I am returning to macOS after some years and am setting up the development environment.

When I used macOS earlier there was only one popular terminal - iTerm2 ( no warp, fig ), on which I installed zsh and Prezto to get up and running.

Now I see many options for working on macOS
 - more editors like warp, fig
 - starship
 - prezto

I would like to know how these three are different ? It seems to me that same functionality can be provided by either of them like prompt customization. Want to know which ones to install. I have seen setups with many combinations of these. 

Feature-wise I would like to have syntax highlighting, prompt customization, reverse string search at least.

**Comment by mickimnet** [comment]
Most of them are programs for different things:

- _Warp_ is a terminal—like _Apple's_ Terminal app, _iTerm_, _Alacritty_ and so on
- _Fig_ extends the functionality of the input line, basically with a graphical autocomplete, suggestions
- _Starship_ show's customizable information in your prompt (which is before the input line)

And as I don't use _Zsh_ as my main shell, after some googling, it looks to me, like _Prezto_ is a fork of _oh-my-zsh_, and therefore touches autocomplete, prompt and suggestions.

I hope that helps.

**Reply by davidkna** [reply]
In general, zsh frameworks like oh-my-zsh and prezto can be used with starship if you disable their respective prompt features.

**Reply by purezen** [reply]
Thanks guys, that does help. Didn't know that fig is not on the same level as let's say iTerm.

Though I am not getting then, what configuration should I go for. I am going to experiment with both iTerm and Warp.

Warp already has some prompt customization. As for iTerm, Starship's customization is reminiscent of Prezto's.

So, it seems to have Starship in iTerm only (?). Also, in what order should I install let's say fig and starship on iTerm ? Is there possibility to mess up by installing them in some order ?

Also, going by the features these tools provide, it seems zsh frameworks like Prezto are the most redundant. Is it so?

**Reply by davidkna** [reply]
Warp uses its own prompt by default, and you will need to enable custom prompt support in the settings. Fig should work with starship, though fig expects to run their script at the start and end of zshrc, so you should keep it that way.
OMZ and prezto aren't quite redundant, they will also enable additional command completions, syntax highlighting and enable features like shell history and other common-sense defaults.

**Reply by purezen** [reply]
Thanks a ton, that should be helpful!
